src/components/PublicRoom.js

```diff
@@ -6,36 +6,53 @@ import MessageList from "./MessageList";
 import SendMessage from "./SendMessage";
 
 import { Message } from "./Message";
+import* as ws from "../socketClient"
 
 const TEMP = [
   new Message(
     "me",
+    "public",
     "HelloWorld Lorem ipsum dolor sit amet, consectetur adipisicing elit. Omnis, ratione fugit. Atque commodi autem velit sit est dolores aut dignissimos, tene",
     "15:30"
   ),
-  new Message("me", "HelloWorld", "15:30"),
+  new Message("me", "public", "HelloWorld", "15:30"),
   new Message(
     "Mr.Lorem",
+    "public",
     "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Omnis, ratione fugit. Atque commodi autem velit sit est dolores aut dignissimos, tenetur, impedit quaerat quo odio aliquam optio reiciendis, in id?",
     "15:30"
   ),
-  new Message("Classic man", "asdsadads", "15:30")
+  new Message("Classic man", "public", "asdsadads", "15.35")
 ];
-
+let sock = ws.connect("ws://35.240.212.170/ws");
 export default class PublicRoom extends React.Component {
   constructor() {
     super();
     this.state = {
-      name: "Public"
-      , messages: TEMP
+      name: "Public",
+      messages: TEMP
+    };
+    sock.onmessage = (msg) => {
+      let some = JSON.parse(msg.data)
+      var time = new Date();
+      console.log(`[info] receive message: ${msg.data}`);
+      TEMP.push(new Message(some.sender, some.destination ,some.message, time.getHours()+":"+time.getMinutes()));
+      this.setState({messages: TEMP});
     };
+    this.sendMessageToSocket = this.sendMessageToSocket.bind(this);
   }
+
+  sendMessageToSocket(message) {
+    let msg = JSON.stringify(new Message("me","public",message,"9000"));
```

Comment:
Hard coded information in Message object
consider using value from existing state/props instead

something like :
```suggestion
    let msg = JSON.stringify(new Message("me", name, message, "9000"));
```
also consider storing username in state so you can use it to construct Message object

Reply:
PS: code format (space between parameter value)